### PR TITLE
Attempts to make CI integration test logs more useful

### PR DESCRIPTION
**Pull request by tcsc**
Actually tracking down the cause of a failure in the integration tests can be hard:

1. It's hard to get an overall summary of what failed
2. The tests sometimes emit no output before timing out, meaning any diagnostic info is lost
3. The emitted logs are too voluminous for a human to parse
4. The emitted logs can present information out of order 
5. It's often hard to tell where the output from one test ends and the next one begins

This patch attempts to address these concerns without attempting to rewrite any of the underlying teleport logging.

- It improves the `render-tests` script to (optionally) report progress per-test, rather than on a per-package basis. My working hypothesis on the tests that time out with no output is that `go test ./integration` is waiting for the entire set of integration tests  tests to be complete before reporting success or failure. Reporting on a per-test cycle gives faster feedback and means that any timed-out builds should give at least _some_ idea of where they are stuck.

- Adds the `render-tests` filter to the `integration` and `integration-root` make targets. This will show an overall summary of test results, as well as 
  - Discarding log output from passing tests to increase signal-to-noise ratio, and 
  - Strongly delimiting the output from each failed test, making failures easier to find.

- Removes the notion of a failure-only logger in favour of post-processing the log events with `render-tests`. The failure-only logger catches log output from the tests and only forwards it to the console if the test fails. Unfortunately, not _all_ log output is guaranteed to pass through this logger  (some `teleport` packages do not honour the configured logger, and reports from the go race detector _certainly_ don't), meaning some output is presented at the time it happens, and other output is batched and displayed at the end of the test. This makes working out what happened where harder than it need be.

In addition, this patch also promotes the `render-tests` script into a fully-fledged program, with appropriate makefile targets, `make clean` support, etc. It is now also more robust in the face on non-JSON output from `go test` (which happens if a package fails to compile).

### CODE REVIEW

**Comment by tcsc** [comment]
See attached [example.log](https://github.com/gravitational/teleport/files/7805831/example.log).

**Review by zmb3 — APPROVED** [review]
LGTM, but please coordinate with @jimbishopp to see how this fits with #9596

**Comment by jimbishopp** [comment]
@tcsc @zmb3 

Thanks for the heads-up.

> LGTM, but please coordinate with @jimbishopp to see how this fits with #9596

I don't think this overlaps much as this is focused on reporting/rendering of integration tests. I closed my PR because I want to increase the scope a bit, but will wait for this PR to be committed before opening a new PR.

> some teleport packages do not honour the configured logger

This is the key piece I'm going to work on. I may not be able to fix all tests, but we'll at least have a framework to move forward with more consistent logging/test configuration.

File: build.assets/tooling/cmd/render-tests/args.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+func (m *reportMode) String() string {
+	switch *m {
+	case byPackage:
+		return byPackageName
+
+	case byTest:
+		return byTestName
+
+	default:
+		log.Fatalf("Unexpected report mode %d", *m)
```
[the remark this answers is not in the document]
Comment (tcsc) [reply build.assets/tooling/cmd/render-tests/args.go]: Done

**Review by timothyb89 — APPROVED** [review]

**Comment by tcsc** [comment]
> >  some teleport packages do not honour the configured logger
> This is the key piece I'm going to work on. 

I think there is some output that will never be able to pass through our logging system (e.g. race detector reports - which was what started me off on this PR in the first place). So while we _definitely_ need a better logging setup, and I will definitely be the first in line to lend a hand if you ask, be wary of assuming that we can catch & categorize _everything_ - especially when running tests.

**Pull request merged**